classes/order/OrderHistory.php

```diff
@@ -136,16 +136,17 @@ public function changeIdOrderState($new_order_state, $id_order, $use_existing_pa
                 $customer = new Customer((int) $order->id_customer);
                 $links = array();
                 foreach ($assign as $product) {
-                    $text = Tools::htmlentitiesUTF8($product['name']);
+                    $complementaryText = '';
                     if (isset($product['deadline'])) {
-                        $text .= '&nbsp;' . $this->trans('expires on %s.', array($product['deadline']), 'Admin.Orderscustomers.Notification');
+                        $complementaryText .= ' ' . $this->trans('expires on %s.', array($product['deadline']), 'Admin.Orderscustomers.Notification');
                     }
                     if (isset($product['downloadable'])) {
-                        $text .= '&nbsp;' . $this->trans('downloadable %d time(s)', array((int) $product['downloadable']), 'Admin.Orderscustomers.Notification');
+                        $complementaryText .= ' ' . $this->trans('downloadable %d time(s)', array((int) $product['downloadable']), 'Admin.Orderscustomers.Notification');
                     }
                     $links[] = array(
-                        'text' => $text,
+                        'text' => Tools::htmlentitiesUTF8($product['name']),
                         'url' => $product['link'],
+                        'complementary_text' => $complementaryText,
```

Comment:
```suggestion
                        'complementary_text' => implode(' ', $complementaryText),
```

And no more spaces :)

Reply:
Or I missed something about @Progi1984 's comment :sweat_smile:

Reply:
I'm pretty sure a space should be put at the begining too otherwise the link will be showed as : [Hummingbird printed sweater]()expires on 2019-10-24. downloadable 10 time(s)
instead of
[Hummingbird printed sweater]() expires on 2019-10-24. downloadable 10 time(s)

Reply:
Maybe this one should be add in the template.